src/main/java/org/fenixedu/academic/ui/struts/action/administrativeOffice/studentEnrolment/StudentEnrolmentsDA.java

```diff
@@ -102,4 +104,44 @@ public ActionForward backViewRegistration(ActionMapping mapping, ActionForm form
         return mapping.findForward("visualizeRegistration");
     }
 
+    public ActionForward annulEnrolment(final ActionMapping mapping, final ActionForm form, final HttpServletRequest request,
+            final HttpServletResponse response) {
+
+        try {
+            atomic(() ->
```

Comment:
I think it would be better to reduce the amount of code inside these atomic blocks. I would extract the `enrolment` variable outside of it, and the just use `atomic(enrolment::annul)`.